go/exorelay/exorelay_test.go

```diff
@@ -71,6 +71,53 @@ func FeatureContext(s *godog.Suite) {
 		}
 		return nil
 	})
+
+	s.Step(`^sending the message "([^"]*)"$`, func(message string) error {
+		return exoInstance.Send(message, nil)
+	})
+
+	s.Step(`^sending the message "([^"]*)" with the payload:$`, func(message string, payloadStr *gherkin.DocString) error {
+		var payload map[string]interface{}
+		err := json.Unmarshal([]byte(payloadStr.Content), &payload)
+		if err != nil {
+			return err
+		}
+		return exoInstance.Send(message, payload)
+	})
+
+	s.Step(`^trying to send an empty message$`, func() error {
+		savedError = exoInstance.Send("", nil)
+		return nil
```

Comment:
In the current Cucumber paradigm, if the step says "When trying to xxx", it is expected that the activity xxx fails. If it succeeds, this step should fail. We can check the exact error in a subsequent step, but here we should check that an error has occurred at all. So, what do you think about adding this line:

```go
if err == nil {
  return fmt.errorf("Expected error, got none")
} else {
  return nil
}
```

I have run into a number of situations where it was expected that an error happens like this, the error didn't occur because of a bug in the code, but the step was still passing in Cucumber. That was always misleading and made debugging harder.